### PR TITLE
Refactoring on Feature Extensions (Followup to Installer Rewrite)

### PR DESCRIPTION
**Pull request by rajeevr2715**
**Requirements**

Address the changes requested in:

1. https://github.com/tmobile/jazz-installer/pull/537#discussion_r285956900
2. https://github.com/tmobile/jazz-installer/pull/539#discussion_r286171544
3. https://github.com/tmobile/jazz-installer/pull/527#discussion_r281252447 

 **Description of the Change**
Update module structure and single entry script for both install/uninstall. Replace arg parse with click module.

### CODE REVIEW

File: features/extensions/apigee/setup.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    prompt=True
+)
+@click.option(
+    '--jenkins_userpass',
+    nargs=2,
+    required=True,
+    help='Provide the username and password \
+    of the jenkins separated by a space',
+    prompt=True)
+def uninstall(region, stackprefix, jazz_userpass, jazz_apiendpoint, jenkins_url, jenkins_userpass):
```
Comment (bleggett) [inline features/extensions/apigee/setup.py]: Personal preference: I find it cleaner to keep the `click` variable declarations in a separate file from the logic that uses them, so there's a clean visual separation between the inputs and the execution. Makes everything easier to read.

I'd recommend moving the `uninstall/install/terraform` "do things" functions to their own file and referencing them from here instead.

File: features/extensions/multiaccount/setup.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    prompt=True
+)
+@click.option(
+    '--jenkins_userpass',
+    nargs=2,
+    required=True,
+    help='Provide the username and password \
+    of the jenkins separated by a space',
+    prompt=True)
+def install(regions, stackprefix, aws_accesskey, aws_secretkey, jazz_apiendpoint,
```
Comment (bleggett) [inline features/extensions/multiaccount/setup.py]: Ditto, see above comment.

File: features/extensions/splunk/setup.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+)
+@click.option(
+    '--jazz_userpass',
+    nargs=2,
+    required=True,
+    help='Provide the username and password \
+    of the jazz application separated by a space',
+    prompt=True
+)
+def install(splunk_endpoint, splunk_token, splunk_index, jazz_apiendpoint, jazz_userpass):
```
Comment (bleggett) [inline features/extensions/splunk/setup.py]: This one only has a single `install` method that is pretty tiny and very few options, so this one doesn't need to be split into multiple files, IMHO.

File: features/utils/jenkins.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            "bash",
+            "utils/"+credfile,
+            "%s" % (jenkins_cli_command),
+            credential_id,
+            key,
+            value
+            ])
+
+
+def startJob(url, username, password, jobUrl):
```
Comment (bleggett) [inline features/utils/jenkins.py]: `requests.post(url)`, not `curl`, please

Comment (rajeevr2715) [reply features/utils/jenkins.py]: Done

File: features/extensions/splunk/setup.py (hunk 3 above)
Comment (rajeevr2715) [reply features/extensions/splunk/setup.py]: Ok

File: features/extensions/multiaccount/setup.py (hunk 2 above)
Comment (rajeevr2715) [reply features/extensions/multiaccount/setup.py]: Done

File: features/extensions/apigee/setup.py (hunk 1 above)
Comment (rajeevr2715) [reply features/extensions/apigee/setup.py]: Done

File: features/extensions/multiaccount/setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rajeevr2715) [reply features/extensions/multiaccount/setup.py]: Yeah, already removed it, https://github.com/tmobile/jazz-installer/pull/541/commits/c0a5353d4c837ffdd5cde1170a4e9a6260af7752

**Pull request merged**